example_test.go

```diff
@@ -0,0 +1,338 @@
+	var t8 = Team{ID: 8, RoomID: 5, Name: "Leadership"}
+	var teams = []Team{t1, t2, t3, t4, t5, t6, t7, t8}
+	for _, t := range teams {
+		err := db.Query(nil, insertTeam, t).Run()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Example 1
+	// Find the team the employee 1 works in.
+	selectSomeoneInTeam := sqlair.MustPrepare(`
+		SELECT &Team.*
+		FROM teams
+		WHERE id = $Employee.team_id`,
+		Employee{}, Team{},
+	)
+
+	// Get returns a single result.
+	var team = Team{}
+	err = db.Query(nil, selectSomeoneInTeam, e1).Get(&team)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%s is on the %s team.\n", e1.Name, team.Name)
+
+	// Example 2
+	// Find out who is in location l1 and what team they work for.
+	selectPeopleInRoom := sqlair.MustPrepare(`
+		SELECT e.* AS &Employee.*, (t.team_name, t.id) AS (&Team.*)
+		FROM employees AS e, teams AS t
+		WHERE t.room_id = $Location.room_id AND t.id = e.team_id`,
+		Employee{}, Team{}, Location{},
+	)
+
+	// GetAll returns all the results.
+	var roomDwellers = []Employee{}
+	var dwellersTeams = []Team{}
+	err = db.Query(nil, selectPeopleInRoom, l1).GetAll(&roomDwellers, &dwellersTeams)
+	if err != nil {
+		panic(err)
+	}
+
+	for i := range roomDwellers {
+		fmt.Printf("%s (%s), ", roomDwellers[i].Name, dwellersTeams[i].Name)
+	}
+	fmt.Printf("are in %s.\n", l1.Name)
+
+	// Example 3
+	// Cycle through employees until we find one in the Penthouse.
+
+	// A map with a key type of string is used to
+	// pass arguments that are not fields of structs.
+	// sqlair.M is of type map[string]any but if
+	// the map has a key type of string it can be used.
+	selectPeopleAndRoom := sqlair.MustPrepare(`
+		SELECT (e.name, t.team_name, l.room_name) AS (&M.employee_name, &M.team, &M.location)
+		FROM locations AS l
+		JOIN teams AS t
+		ON t.room_id = l.room_id
+		JOIN employees AS e
+		ON e.team_id = t.id`,
+		sqlair.M{},
+	)
+
+	// Results can be iterated through with an Iterable.
+	// iter.Next prepares the next result.
+	// iter.Get reads it into structs.
+	// iter.Close closes the query returning any errors. It must be called after iteration is finished.
+	iter := db.Query(nil, selectPeopleAndRoom).Iter()
+	defer iter.Close()
+	for iter.Next() {
+		var m = sqlair.M{}
+		err := iter.Get(&m)
+		if err != nil {
+			panic(err)
+		}
+		if m["location"] == "The Penthouse" {
+			fmt.Printf("%s from team %s is in %s.\n", m["employee_name"], m["team"], m["location"])
+			break
+		}
+	}
+	err = iter.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	drop := sqlair.MustPrepare(`
+		DROP TABLE employees;
+		DROP TABLE teams;
+		DROP TABLE locations;`,
+	)
+	err = db.Query(nil, drop).Run()
+	if err != nil {
+		panic(err)
+	}
+
+	// Output:
+	//Alastair is on the Engineering team.
+	//Alastair (Engineering), Ed (Engineering), Marco (Engineering), Pedro (Management), Serdar (Presentation Engineering), Lina (Presentation Engineering), are in The Basement.
+	//Gustavo from team Leadership is in The Penthouse.
```

Comment:
[Nitpick] I think that whitespace is trimmed from output comments, so we could make it more readable with some spacing in the beginning.

Reply:
Changed